python/external_config/config/config_base.py

```diff
@@ -201,7 +226,21 @@ def request_commands(self, project_id, entity_type, entity_id, link_entity_type,
         :param str engine_fallback: If the main engine isn't available for the given
             entity id and project, request generate commands for the fallback engine
             specified. This can be useful in backwards compatibility scenarios.
-        """
+
+        :raises: RuntimeError if this configuration's status does not allow for
+            commands requests.
+        """
+        # Make sure that we're a valid configuration. If we aren't, then we
+        # won't be able to successfully request commands. That being the case,
+        # we can raise here.
+        if not self.is_valid:
```

Comment:
I'm having second thoughts about this, @manneohrstrom. I'm wondering if status should just be a property on the InvalidExternalConfiguration implementation, which itself raises if commands are requested from it. Then we can remove the status from the base external config entirely. Thoughts?

Reply:
We'd then also no longer need the CONFIGURATION_READY status. In fact, we might not need the status enum at all since we only have one case right now. We could always add status back in the future if we end up with multiple possibilities for why a config is invalid. That would cut out some arguably-premature optimization here.